### PR TITLE
add random tube phantom (with -N <num>)

### PR DESCRIPTION
**Pull request by jtamir**
Adds the ability to make a tube phantom with random number of tubes. Also supports basis.

Example:  
`bart phantom -N 100  -r 123  -x 256 phn`
produces
image

### CODE REVIEW

File: src/simu/phantom.c [hunk 1]
```diff
@@ -517,18 +518,98 @@ static void separate_bckgrd(int Nb, struct ellipsis_s bckgrd[Nb], int Nf, struct
...
+			for (int i = 0; i < 2 * N - 2; i+=2) {
+
+				double sx = 0;
+				double sy = 0;
+				double px = 0;
+				double py = 0;
+
+				bool overlap = true;
+
+				while (overlap) {
```
Comment (uecker) [inline src/simu/phantom.c:540]: Doesn't this go into an endless loop when N is too high and there is not enough space?

**Comment by jtamir** [comment]
It can happen. I have a version that shrinks the tubes over time until they fit. Should I add a graceful fail at n<N tubes if runtime exceeds some number?

Jon
www.jtsense.com
________________________________

File: src/simu/phantom.c (hunk 1 above)
Comment (uecker) [reply src/simu/phantom.c:540]: I would just error out after a reasonable number of trials.

**Comment by jtamir** [comment]
I added some checks and changed the way ellipses are added over time. It will now error if it isn't able to find a solution. Nonetheless, with the change I was able to generate 10,000 ellipses (see screenshot):
image

**Comment by uecker** [comment]
I will merge this, it is a nice feature! But as these are all spheres at the moment and not ellipses, the check for overlap should be simple, or I am missing something?

**Comment by jtamir** [comment]
I wanted to leave it in so that at some point it could be extended, but I think simpler is better. So I refactored, fixed some bug with the borders, and made it explicitly for circles only.

Here's 10,000 tubes, with fixes
image

**Pull request merged**